tests/testthat/setup-options.R

```diff
@@ -0,0 +1,17 @@
+opts_partial_match_old <- list(
+  warnPartialMatchDollar = getOption("warnPartialMatchDollar"),
+  warnPartialMatchArgs = getOption("warnPartialMatchArgs"),
+  warnPartialMatchAttr = getOption("warnPartialMatchAttr")
+)
```

Comment:
Do we need to keep the old options? They are never restored as `withr` handles that on its own.

Reply:
This was left for possible exclusions when I was mass copy&paste. This package (among others) does not have exclusions so this is actually redundant (contrary to e.g. [tern](https://github.com/insightsengineering/tern/pull/1182)). I honestly don't know. I'm fine with removing this from here.

Reply:
Ok, it seems fine to leave, however a comment giving some context on how to use might be nice to have.

Reply:
```suggestion
# May be used during tests to revert to non-strict environment
opts_partial_match_old <- list(
  warnPartialMatchDollar = getOption("warnPartialMatchDollar"),
  warnPartialMatchArgs = getOption("warnPartialMatchArgs"),
  warnPartialMatchAttr = getOption("warnPartialMatchAttr")
)
```